### PR TITLE
[Updates]

### PR DESCRIPTION
**Pull request by r-scheele**
- Renamed files, functions, methods, variables, and classes to mirror what they do; For example, a file named github.py should have functions such as initialize_github, etc.
So basically, let's update the application code structure to match their use.
- Github action updates; Added jobs to handle pytest coverage and sort all module imports in the project directory
-  Adjusted `/data` route to return the group name as a child of the schema. For instance; `schema.usergroups` as the key, instead of `usergroups`

### CODE REVIEW

File: app/server/services/github.py [hunk 1]
```diff
@@ -39,9 +53,14 @@ def initialize(self) -> None:
 
         self.repo_git_path = initialized_repo.git_dir
 
-    def push(self) -> None:
+    def github_push(self) -> None:
```
Comment (Youngestdev) [inline app/server/services/github.py:56]: Hm. I thought I asked you to rename this back to `push`?

File: app/server/services/github.py [hunk 2]
```diff
@@ -23,10 +34,13 @@ def __init__(self, repo_url: str, access_token: str, username: str) -> None:
         self.local_repo_path = f"{default_path}/{self.repo_name}"
         self.repo_git_path = ""
 
-    def initialize(self) -> None:
-        """
-        Check if the remote repository is valid and clone the remote repository.
+    def initialize_github(self) -> None:
```
Comment (Youngestdev) [inline app/server/services/github.py:37]: rename this to `initialize`.

**Review by Youngestdev — CHANGES_REQUESTED** [review]

File: app/server/services/github.py (hunk 1 above)
Comment (r-scheele) [reply app/server/services/github.py:56]: Oh, I thought I changed it. I'll fix it now

File: app/server/services/github.py (hunk 2 above)
Comment (r-scheele) [reply app/server/services/github.py:37]: Alright 👍

**Pull request merged**